topdown/eval.go

```diff
@@ -3662,11 +3739,19 @@ func refContainsNonScalar(ref ast.Ref) bool {
 }
 
 func suppressEarlyExit(err error) error {
-	ee, ok := err.(*earlyExitError)
-	if !ok {
-		return err
+	if ee, ok := err.(*earlyExitError); ok {
+		return ee.prev
+	} else if oee, ok := err.(*outerEarlyExitError); ok {
+		return oee.prev
 	}
-	return ee.prev // nil if we're done
+	return err
+}
```

Comment:
Wouldn't `suppressEarlyExit(nil)` also just return `nil`? So there doesn't seem to be a need for the helper below...? 💭

Reply:
The purpose of the `withSuppressEarlyExit` helper is just to make it convenient to wrap a block of code containing a lot of return points without larger re-writing, not to correct any `nil` handling in `suppressEarlyExit`.